src/ClientHandle.cpp

```diff
@@ -264,6 +286,58 @@ void cClientHandle::Kick(const AString & a_Reason)
 
 
 
+bool cClientHandle::BungeeAuthenticate()
+{
+	if (!m_ProxyConnection && cRoot::Get()->GetServer()->OnlyAllowBungeeCord())
+	{
+		Kick("You can only connect to this server using a Proxy.");
+
+		return false;
+	}
+
+	cServer * Server = cRoot::Get()->GetServer();
+
+	// Proxy Forward Secret Check (BungeeGuard)
+	const AString & ForwardSecret = Server->GetProxyForwardSecret();
```

Comment:
Is this a standardised name for this secret? It would make much more sense to call this a "ProxySharedSecret" rather than a forward secret - which makes me think the secret is somehow temporal and does not bring to mind forwarding at all.